lucene/sandbox/src/java/org/apache/lucene/document/LatLonShapeQuery.java

```diff
@@ -163,17 +186,24 @@ public Scorer get(long leadCost) throws IOException {
         };
       }
 
-      /** get a scorer supplier for all other queries (DISJOINT, WITHIN) */
+      /** get a scorer supplier for all other queries (DISJOINT, WITHIN, CONTAINS) */
       protected ScorerSupplier getScorerSupplier(LeafReader reader, PointValues values, Weight weight, ScoreMode scoreMode) throws IOException {
         if (queryRelation == QueryRelation.INTERSECTS) {
           return getIntersectScorerSupplier(reader, values, weight, scoreMode);
         }
-        //For within and disjoint we need two passes to remove false positives in case of multi-shapes.
         FixedBitSet within = new FixedBitSet(reader.maxDoc());
         FixedBitSet disjoint = new FixedBitSet(reader.maxDoc());
-        IntersectVisitor withinVisitor = getDenseIntersectVisitor(within, disjoint, QueryRelation.WITHIN);
-        IntersectVisitor disjointVisitor = getDenseIntersectVisitor(within, disjoint, QueryRelation.DISJOINT);
-        return new RelationScorerSupplier(values, withinVisitor, disjointVisitor, queryRelation) {
+        IntersectVisitor visitor;
+        IntersectVisitor disjointVisitor;
+        if (queryRelation == QueryRelation.CONTAINS) {
+          visitor = getDenseIntersectVisitor(within, disjoint, queryRelation);
```

Comment:
the `within`/`disjoint` naming of the bitsets feels a bit weird now (my fault, sorry!). Maybe `candidates`/`rejected` or something like that instead?